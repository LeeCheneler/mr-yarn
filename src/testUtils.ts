import { exec as doExec, ExecOptions } from 'child_process'
import { promisify } from 'util'

/**
 * Simply promisifies the standard node 'exec' function
 * @param options options
 */
export const exec = async (command: string, options: ExecOptions) => promisify(doExec)(command, options)
