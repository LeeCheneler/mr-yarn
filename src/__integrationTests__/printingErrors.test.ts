import { resolve } from 'path'
import { exec } from '../testUtils'

/**
 * Fixtures directory containing variants of workspace setup to assert against
 */
const fixturesDir = resolve(process.cwd(), 'src/__integrationFixtures__')

it('Should print an errors', async () => {
  const { stdout } = await exec('mr add deep-freeze', { cwd: resolve(fixturesDir, 'invalid') })

  expect(stdout).toMatchSnapshot()
})
