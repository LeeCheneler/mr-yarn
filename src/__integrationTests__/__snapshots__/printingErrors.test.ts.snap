// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should print an errors 1`] = `
"[35mMr Yarn[39m [31merror[39m: Missing { \\"private\\": false } in 'package.json'
"
`;
